src/infrastructure.jl

```diff
@@ -0,0 +1,322 @@
+################################################################################
+
+@enum RenderType SVG PNG PDF EPS
+
+"The Cairo backend object.  Used to dispatch to CairoMakie methods."
+struct CairoBackend <: AbstractPlotting.AbstractBackend
+    typ::RenderType
+    path::String
+end
+
+"""
+    struct CairoScreen{S} <: AbstractScreen
+A "screen" type for CairoMakie, which encodes a surface
+and a context which are used to draw a Scene.
+"""
+struct CairoScreen{S} <: AbstractPlotting.AbstractScreen
+    scene::Scene
+    surface::S
+    context::CairoContext
+    pane::Nothing # TODO: GtkWindowLeaf
+end
+
+
+function CairoBackend(path::String)
+    ext = splitext(path)[2]
+    typ = if ext == ".png"
+        PNG
+    elseif ext == ".svg"
+        SVG
+    elseif ext == ".pdf"
+        PDF
+    elseif ext == ".eps"
+        EPS
+    else
+        error("Unsupported extension: $ext")
+    end
+    CairoBackend(typ, path)
+end
+
+# we render the scene directly, since we have
+# no screen dependent state like in e.g. opengl
+Base.insert!(screen::CairoScreen, scene::Scene, plot) = nothing
+
+function Base.show(io::IO, ::MIME"text/plain", screen::CairoScreen{S}) where S
+    println(io, "CairoScreen{$S} with surface:")
+    println(io, screen.surface)
+end
+
+# Default to ARGB Surface as backing device
+# TODO: integrate Gtk into this, so we can have an interactive display
+"""
+    CairoScreen(scene::Scene; antialias = Cairo.ANTIALIAS_BEST)
+Create a CairoScreen backed by an image surface.
+"""
+function CairoScreen(scene::Scene; device_scaling_factor = 1, antialias = Cairo.ANTIALIAS_BEST)
+    w, h = round.(Int, scene.camera.resolution[] .* device_scaling_factor)
+    surf = Cairo.CairoARGBSurface(w, h)
+
+    # this sets a scaling factor on the lowest level that is "hidden" so its even
+    # enabled when the drawing space is reset for strokes
+    # that means it can be used to increase or decrease the image resolution
+    ccall((:cairo_surface_set_device_scale, Cairo.libcairo), Cvoid, (Ptr{Nothing}, Cdouble, Cdouble),
+        surf.ptr, device_scaling_factor, device_scaling_factor)
+
+    ctx = CairoContext(surf)
+    Cairo.set_antialias(ctx, antialias)
+
+    return CairoScreen(scene, surf, ctx, nothing)
+end
+
+
+"""
+    CairoScreen(
+        scene::Scene, path::Union{String, IO};
+        mode = :svg, antialias = Cairo.ANTIALIAS_BEST
+    )
+Creates a CairoScreen pointing to a given output path, with some rendering type defined by `mode`.
+"""
+function CairoScreen(scene::Scene, path::Union{String, IO}, mode::Symbol; device_scaling_factor = 1, antialias = Cairo.ANTIALIAS_BEST)
+
+    # the surface size is the scene size scaled by the device scaling factor
+    w, h = round.(Int, scene.camera.resolution[] .* device_scaling_factor)
+
+    if mode == :svg
+        surf = CairoSVGSurface(path, w, h)
+    elseif mode == :pdf
+        surf = CairoPDFSurface(path, w, h)
+    elseif mode == :eps
+        surf = Cairo.CairoEPSSurface(path, w, h)
```

Comment:
I guess the `Cairo.` is necessary?

Reply:
We should actually just not import all this stuff, since it isn't used anywhere but here.